Resources/public/js/views/ez-subitemlistview.js

```diff
@@ -160,6 +173,164 @@ YUI.add('ez-subitemlistview', function (Y) {
+         * Hide the error icon.
+         *
+         * @method _hideErrorIcon
+         * @protected
+         * @param {Object} input the DOM node of the input
+         * @param {Object} selectedCell the DOM node of the current selected cell
+         */
+        _hideErrorIcon: function (input, selectedCell) {
+            //input.removeClass('ez-subitem-error-input');
+            selectedCell.removeClass('ez-subitem-error-priority-cell');
+            selectedCell.addClass('ez-subitem-selected-priority-cell');
+        },
+
+        /**
+         * Show the 'validate' & 'cancel' buttons which allow to update or not the priority.
+         *
+         * @method _displayPriorityButtons
+         * @protected
+         * @param {Object} e event facade
+         */
+        _displayPriorityButtons: function (e) {
+            var selectedCell = this.get('container').one('#priority-cell-' + e.target.getAttribute('data-location-id'));
+
+            e.preventDefault();
+            if (!this.get('isOnePrioritySelected')) {
+                this._set('isOnePrioritySelected', true);
+                selectedCell.removeClass('ez-subitem-hovered-priority-cell');
+                selectedCell.addClass('ez-subitem-selected-priority-cell');
+                e.target.removeAttribute('readonly');
+            }
+        },
+
+        /**
+         * Hide the 'validate' & 'cancel' buttons which allow to update or not the priority.
+         * Also fill back priority input with the current sub item priority.
+         *
+         * @method _hidePriorityButtons
+         * @protected
+         * @param {Object} e event facade
+         */
+        _hidePriorityButtons: function (e) {
+            var selectedCell = this.get('container').one('.ez-subitem-selected-priority-cell'),
+                input = this.get('container').one('#' + e.target.getAttribute('data-priority-input'));
+
+            Y.Array.each(this.get('subitems'), Y.bind(function (locStruct) {
+                if (locStruct.location.get('locationId') == input.getAttribute('data-location-id')) {
+                    input.set('value', locStruct.location.get('priority'));
+                    input.setAttribute('readonly', 'readonly');
+                }
+            },this));
+
+            e.preventDefault();
+            selectedCell.removeClass('ez-subitem-selected-priority-cell');
+            this._set('isOnePrioritySelected', false);
+        },
+
+        /**
+         * Event handler allowing to change the priority of the subitem
+         *
+         * @method _setPriority
+         * @protected
+         * @param {Object} e event facade
+         */
+        _setPriority: function (e) {
+            var button = e.target,
+                selectedCell = this.get('container').one('.ez-subitem-selected-priority-cell'),
+                input;
+
+            e.preventDefault();
+            input = this.get('container').one('#' + button.getAttribute('data-priority-input'));
```

Comment:
this is also repeated, you should introduce a method like `_getPriorityInput: function (button)`